src/sentry/tasks/weekly_escalating_forecast.py

```diff
@@ -30,20 +32,29 @@ def run_escalating_forecast() -> None:
     Run the escalating forecast algorithm on archived until escalating issues.
     """
     logger.info("Starting task for sentry.tasks.weekly_escalating_forecast.run_escalating_forecast")
-    # TODO: Do not limit to project id = 1 and limit 10 once these topics are clarified
-    # TODO: If possible, fetch group_id instead of the entire group model
-    until_escalating_groups = list(
-        Group.objects.filter(
-            status=GroupStatus.IGNORED,
-            substatus=GroupSubStatus.UNTIL_ESCALATING,
-            project__id=1,
-        )[:10]
-    )
-    logger.info(
-        "Checking for archived until escalating groups",
-        extra={"num_groups": len(until_escalating_groups)},
-    )
-    if not until_escalating_groups:
-        return
-
-    generate_and_save_forecasts(until_escalating_groups)
+
+    for project_ids in RangeQuerySetWrapper(
+        Project.objects.filter(status=ObjectStatus.VISIBLE).values_list("id", flat=True),
+        result_value_getter=lambda item: item,
+        step=10000,
+    ):
+        generate_forecasts_for_projects.delay(project_ids=[project_ids])
+
+
+@instrumented_task(
+    name="sentry.tasks.weekly_escalating_forecast.generate_forecasts_for_projects",
+    queue="weekly_escalating_forecast",
+    max_retries=3,
+    default_retry_delay=60,
+)  # type: ignore
+def generate_forecasts_for_projects(project_ids: List[int]) -> None:
+    for project_id in project_ids:
+        for until_escalating_groups in RangeQuerySetWrapper(
```

Comment:
Just noting that since we use `RangeQuerySetWrapper` here we might break index usage. This wrapper works by sorting on `id` and also filtering on id ranges to manage pages. So for any projects that have a lot of groups it might cause slow queries.

Since the id is in the index, this might work ok, but I'm not sure how well sorting by id will work since it's the last column in the index. Maybe we don't need to address right now, but see how well it performs in prod

Reply:
What's the benefit of using `RangeQuerySetWrapper`?

Reply:
It brings models into memory in batches, so that if you're iterating over say, 1mm groups, you don't bring 1mm groups into memory then hit an OOM. Instead, it will pull batches of 10k at a time. It does this by sorting by `id`, then for each page, it'll say `id > <max id from last page>`